agnostic-angular/libs/ag/src/lib/menu.component.stories.ts

```diff
@@ -0,0 +1,146 @@
+import {
+  componentWrapperDecorator,
+  moduleMetadata,
+  Story,
+} from '@storybook/angular';
+import { Meta } from '@storybook/angular/types-6-0';
+import { AgModule } from './ag.module';
+import { MenuComponent } from './menu.component';
+import { action } from '@storybook/addon-actions';
+
+export default {
+  title: 'AG—Angular (Beta)/Menu',
+  component: MenuComponent,
+  decorators: [
+    // Cannot get preview.js or global decorator solutions to work.
+    // https://storybook.js.org/tutorials/intro-to-storybook/angular/en/composite-component/
+    componentWrapperDecorator(
+      (story) =>
+        `<div style="font-family: var(--agnostic-font-family-body); line-height: 1.4; font-size: var(--agnostic-body);">${story}</div>`
+    ),
+    moduleMetadata({
+      imports: [AgModule],
+    }),
+  ],
+} as Meta;
+
+const actions = {
+  onSelectionChange: action('onSelectionChange'),
+  onTriggerClick: action('onTriggerClick'),
+  onItemClick: action('onItemClick'),
+  onItemKeydown: action('onItemKeydown'),
+};
+
+const Template: Story<MenuComponent> = (args: MenuComponent) => ({
+  props: {
+    action: {
+      onSelectionChange: actions.onSelectionChange,
+      onTriggerClick: actions.onTriggerClick,
+      onItemClick: actions.onItemClick,
+    },
+    closeOnSelect: args.closeOnSelect == null ? true : args.closeOnSelect,
+    closeOnClickOutside:
+      args.closeOnClickOutside == null ? false : args.closeOnClickOutside,
+    icon: args.icon || '▾',
+    isBordered: args.isBordered == null ? false : args.isBordered,
+    isRounded: args.isRounded == null ? false : args.isRounded,
+    id: args.id || 'sup',
+    menuTitle: args.menuTitle || 'Players',
+    size: args.size || '',
+    selectedMenuItem: null,
+  },
+  template: `
+  <ag-menu
+    class="my-menu"
+    menuTitle="Default Menu"
+    (triggerClick)="action.onTriggerClick($event)"
+    (selectionChange)="action.onSelectionChange($event)"
+  >
+    <!-- Mark the Template for the menu components -->
+    <ng-template agMenuContent>
+      <button
+        ag-menu-item
+        *ngFor="let item of ['Item 1', 'Item 2', 'Item 3']; let idx = index"
+        [isSelected]="selectedMenuItem === idx"
+        [disabled]="false"
+        (keydown)="action.onItemKeydown($event)"
+        (click)="selectedMenuItem = idx; action.onItemClick($event)"
+      >
+        {{ item }} 😅😅
+      </button>
+    </ng-template>
+  </ag-menu>
+`,
+});
+
+// Should use default layout/config
+export const All = Template.bind({});
+All.args = {
+  closeOnSelect: true,
+  closeOnClickOutside: true,
+  icon: '▾',
+  id: 'agnosticUiToTheMoon',
+  isBordered: false,
+  isRounded: false,
+  menuTitle: 'Default Menu Appearance',
+  size: '',
+};
+
+export const noCloseOnOutsideClick = Template.bind({});
+noCloseOnOutsideClick.args = {
+  closeOnClickOutside: false,
+  id: 'noClose',
+  menuTitle: 'Players',
+};
+
+export const noCloseOnSelect = Template.bind({});
+noCloseOnSelect.args = {
```

Comment:
I'm seeing that in this story spec we're also for some reason NOT closing on click outside. We should only NOT close on clicking the menu option. Can't really show a screenshot but this is the story:

![image](https://user-images.githubusercontent.com/142403/163710082-5942aa88-ac81-40e2-94a1-f55f2f865e37.png)

Ideally we'd

Reply:
I'm not quite sure why this isn't working because I think `closeOnClickOutside: true` is the default and
https://github.com/AgnosticUI/agnosticui/blob/34e7e6b4a137a13c36bc01b0a931d93cfee31172/agnostic-angular/libs/ag/src/lib/menu.component.ts#L273
logic seems to be correct 🤷‍♂️ 

I'll add this to my follow ups issue which is a small list of things that can be tackled on a follow up iteration

Reply:
I'll have to take another look at this. It could simply be something in how the story is implemented. Once I got the "All" variant wired up to the storybook UI, I was mostly testing in there. We should get the variants all configured correctly and working correctly, though. Comparing Storybook across frameworks was super helpful as a first-time contributor